### PR TITLE
Add new typeing for System Task

### PR DESCRIPTION
**Pull request by leocwlam**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## system-task ([<kbd>unpkg</kbd>](https://unpkg.com/browse/system-task@latest/))
was missing the following properties:
1. SyncProcess
1. AsyncProcess
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4a001673eb4b72b94c274d6e2ad00428b52195b5
</p>

File: types/system-task/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// Definitions by: Leo Lam <https://github.com/leocwlam>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.1
+
+/// <reference types="node" />
+
+export function asyncProcess(items: any[], executeAsyncCall: any, task: SystemTask, errors: any[]): Promise<any>;
+export function syncProcess(items: any[], executeSyncCall: any, task: SystemTask, errors: any[]): Promise<any>;
+
+export default class SystemTask {
```
Comment (DanielRosenwasser) [inline types/system-task/index.d.ts:12]: This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Comment (DanielRosenwasser) [inline types/system-task/index.d.ts:12]:
```suggestion
export = SystemTask;
class SystemTask {
```

Comment (leocwlam) [reply types/system-task/index.d.ts:12]: I have the new PR with your suggestion.
Thanks again.

**Comment by leocwlam** [comment]
Thanks

**Comment by leocwlam** [comment]
I have question-related to release npmjs.com.
image

I think it is missing 
**Installation**, **Summary**, **Details**, **Credits** sessions


I have some other types as the following:
image

**Pull request merged**